custom_components/myskoda/switch.py

```diff
@@ -496,3 +501,113 @@ async def async_turn_on(self, **kwargs):  # noqa: D102
 
     def required_capabilities(self) -> list[CapabilityId]:
         return [CapabilityId.AIR_CONDITIONING_SMART_SETTINGS]
+
+
+class DepartureTimerSwitch(MySkodaSwitch):
+    """Base class for departure timers, handling common functionality."""
+
+    def __init__(self, coordinator, vin, timer_id: int, **kwargs):
+        """Initialize the departure timer switch."""
+        super().__init__(coordinator, vin)  # Initialize parent class (MySkodaEntity)
+        self.timer_id = timer_id  # Store the specific timer ID for each subclass
+
+    def get_timer(self) -> Optional[DepartureTimer]:
+        """Retrieve the specific departure timer by ID."""
+        if departure_info := self.vehicle.departure_info:
+            if departure_info.timers:
+                return next(
+                    (
+                        timer
+                        for timer in departure_info.timers
+                        if timer.id == self.timer_id
+                    ),
+                    None,
+                )
+        return None
+
+    @property
+    def is_on(self) -> bool | None:
+        """Check if the timer is enabled."""
+        if timer := self.get_timer():
+            return timer.enabled
+        return None
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
Storing things in `extra_state_attributes` bring some side effect. For instance, this is never picked up by the `recorder` and this does not have any history.

Are the `extra_state_attributes` the right place for this, or could we maybe use a `schedule` helper?

Reply:
yeah I know, just wanted to put it somewhere so users can actually see what is inside the timer they are enabling.. Its only for information purposes. In future I'm planning to implement also a service call so users can actually configure their own timer from our integration but not today 😄